src/EmotiBitWiFiHost.cpp

```diff
@@ -59,73 +98,84 @@ void EmotiBitWiFiHost::getAvailableSubnets() {
 		{
-			vector<string> ipSplit = ofSplitString(ips.at(subnet), ".");
-			string tempSubnet = ipSplit.at(0) + "." + ipSplit.at(1) + "." + ipSplit.at(2);
-				if (ofFind(availableSubnets, tempSubnet) == availableSubnets.size()) {
-					availableSubnets.push_back(tempSubnet);
-				}
+			vector<string> ipSplit = ofSplitString(ips.at(network), ".");
+			string tempNetwork = ipSplit.at(0) + "." + ipSplit.at(1) + "." + ipSplit.at(2);
+			if (ofFind(availableNetworks, tempNetwork) == availableNetworks.size()
+				&& isInNetworkIncludeList(tempNetwork) 
+				&& !isInNetworkExcludeList(tempNetwork)) {
+					availableNetworks.push_back(tempNetwork);
+			}
 		}
 	}
-	if (availableSubnets.size() != currentavailableSubnets.size()) { //print all subnets whenever new subnets are detected
-		string allAvailableSubnets;
-		for (int subnet = 0; subnet < availableSubnets.size(); subnet++) {
-			allAvailableSubnets += "[" + availableSubnets.at(subnet) + ".*] ";
+	if (availableNetworks.size() != currentavailableNetworks.size()) { //print all Networks whenever new Networks are detected
+		string allAvailableNetworks;
+		for (int network = 0; network < availableNetworks.size(); network++) {
+			allAvailableNetworks += "[" + availableNetworks.at(network) + ".*] ";
 		}
-		ofLogNotice() << "All Subnet(s): " << allAvailableSubnets;
+		ofLogNotice() << "All Network(s): " << allAvailableNetworks;
 	}
 }
 
-void EmotiBitWiFiHost::pingAvailableSubnets() {
-	getAvailableSubnets(); // Check if new subnet appeared after oscilloscope was open (e.g. a mobile hotspot)
+void EmotiBitWiFiHost::pingAvailableNetworks() {
+	getAvailableNetworks(); // Check if new network appeared after oscilloscope was open (e.g. a mobile hotspot)
 	string packet = EmotiBitPacket::createPacket(EmotiBitPacket::TypeTag::HELLO_EMOTIBIT, advertisingPacketCounter++, "", 0); 
 	string ip;
-	if (enableBroadcast) { // ToDo: add enableBroadcast case if it's determined to be desirable
-		advertisingCxn.SetEnableBroadcast(true);
-	}
-	if (emotibitSubnets.size() == 0) { //initial search through all subnets
-		for (int sub = 0; sub < availableSubnets.size(); sub++) {
-			if (enableBroadcast) { 
-				ip = availableSubnets.at(sub) + "." + ofToString(255);
+
+	if (emotibitNetworks.size() == 0) { //initial search through all Networks
+		for (int network = 0; network < availableNetworks.size(); network++) {
+			if (_hostAdvSettings.enableBroadcast) {
+				advertisingCxn.SetEnableBroadcast(true);
+				ip = availableNetworks.at(network) + "." + ofToString(255);
 				advertisingCxn.Connect(ip.c_str(), advertisingPort);
 				advertisingCxn.Send(packet.c_str(), packet.length());
 			}
 			else {
-				for (int hostId = 1; hostId < 255; hostId++) {
-					ip = availableSubnets.at(sub) + "." + ofToString(hostId);
+				advertisingCxn.SetEnableBroadcast(false);
+				for (int hostId = _hostAdvSettings.unicastIpRange.first; hostId < _hostAdvSettings.unicastIpRange.second; hostId++) {
```

Comment:
should be hostId <`=` _hostAdvSettings.unicastIpRange.second

Reply:
It was discussed in our meeting, but i am flagging it in the review as well, just so we don't forget to update that among other things